### ISSUE TITLE
Version 1.0

### ISSUE THREAD

**Issue by bryanjos** [bryanjos/joken#13]
What would be needed to make this 1.0?

**Comment by stephenmuss** [comment]
Would be nice to see RSxxx support. Are there any plans for this?

**Comment by bryanjos** [comment]
A couple of weeks ago someone on the Elixir IRC channel said they were working on it and readying a PR, but I haven't heard back from them. Yes, it's planned. I haven't looked into it yet personally.

**Comment by potatosalad** [comment]
I'm working on a library that implements the majority of the JSON Object Signing and Encryption (JOSE) RFC specifications which might pair up nicely with joken: [potatosalad/erlang-jose](https://github.com/potatosalad/erlang-jose).

The JWT RFC and testing are still unfinished, but all of the [JWS algorithms](https://github.com/potatosalad/erlang-jose#json-web-signature-jws-rfc-7515) are supported. Below is an example of using RS256.

_optional:_ Generate an RSA private key with `openssl genrsa -out rsa.pem 2048`

Using `iex`, you can do the following:

```elixir
private_jwk = :jose_jwk.from_pem_file("rsa.pem")
public_jwk  = :jose_jwk.to_public(private_jwk)

payload = :jsx.encode(%{ username: "johndoe" })

json_serialization = :jose_jwk.sign(payload, %{ "alg" => "RS256", "typ" => "JWT" }, private_jwk)
# {%{alg: :jose_jws_alg_rsa_pkcs1_v1_5},
# %{"payload" => "eyJ1c2VybmFtZSI6ImpvaG5kb2UifQ",
#   "protected" => "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9",
#   "signature" => "VgJ0XWw2SkyEmfUip6Bom8zqbzMsxYzarne3AGQpp4Zrw3IUt-cy7VyJm9bPDMcuWDd9t8Fc3GRnTgYZ4BNGddlgB2V_F2hvi7q-DRlrWlKktr7JaA5nU3csFDRZN5xqqCm4L-tejwv6fmQ1_rrT2lCSsqMNCrLBgIMwe_wvLXrSgWOI-_8-zdVD4-Pwrmh05sNbCGjLFodujQSSvVm83U6V1aEgobtLy9S_ggOHt5m11y5M71dMMnS1sWBZccnPV86W_qaZA11mQNbWsTnn0dovvRNLeoIZdwnn-Jthz-_zoXIwnVFTViXEDRSjxSS806oZAldW7bvaDZi2xEu2PA"}}

compact_serialization = :jose_jws.compact(json_serialization)
# {%{alg: :jose_jws_alg_rsa_pkcs1_v1_5},
# "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJ1c2VybmFtZSI6ImpvaG5kb2UifQ.VgJ0XWw2SkyEmfUip6Bom8zqbzMsxYzarne3AGQpp4Zrw3IUt-cy7VyJm9bPDMcuWDd9t8Fc3GRnTgYZ4BNGddlgB2V_F2hvi7q-DRlrWlKktr7JaA5nU3csFDRZN5xqqCm4L-tejwv6fmQ1_rrT2lCSsqMNCrLBgIMwe_wvLXrSgWOI-_8-zdVD4-Pwrmh05sNbCGjLFodujQSSvVm83U6V1aEgobtLy9S_ggOHt5m11y5M71dMMnS1sWBZccnPV86W_qaZA11mQNbWsTnn0dovvRNLeoIZdwnn-Jthz-_zoXIwnVFTViXEDRSjxSS806oZAldW7bvaDZi2xEu2PA"}

{true, ^payload, _jws} = :jose_jwk.verify(json_serialization, public_jwk)
{true, ^payload, _jws} = :jose_jwk.verify(compact_serialization, public_jwk)
```

**Comment by bryanjos** [comment]
@potatosalad that looks awesome! I will take a look when I get a chance. It might make a good fit.

**Comment by cs-victor-nascimento** [comment]
With all the algorithms now properly implemented (thanks to erlang-jose!!!) and a whole new API, I think we could tag 0.16.0 as a 1.0.0-rc1. 

We should seek feedback from the mailing list and core members. The only thing to call a 1.0.0 release would be, IMHO, a good set of samples. I was thinking of coming up with a simple plug application that can be deployed to Heroku or something like that. But that is more like a plus than a showstopper for 1.0.0. 

@bryanjos  what do you think?

**Comment by bryanjos** [comment]
I think that's a good plan. I agree we should definitely should get feedback from the mailing list and anybody who has built a library around it. Sounds like we are close!

**Comment by bryanjos** [comment]
@cs-victor-nascimento just published 1.0!
https://hex.pm/packages/joken
